tests/Unit/SccmIniFIle.tests.ps1

```diff
@@ -0,0 +1,322 @@
+            {
+                Context "When Set-TargetResource runs successfully with $($test.IniFileName)" {
+                    It 'Should not throw with minmual parameters' {
+                        Set-TargetResource @test
+                        Get-Item -LiteralPath "$IniFilePath/$IniFileName" | Should -Not -Be $null
+                    }
+
+                    It 'Should not throw with optional parameters' {
+                        $combined = $test + $optionalParamatersAll
+                        Set-TargetResource @combined
+                        Get-Item -LiteralPath "$IniFilePath/$IniFileName" | Should -Not -Be $null
+                    }
+
+                    if ($test.iniFileName -eq 'InstallPrimary.ini')
+                    {
+                        It 'Should not throw with all optional parameters for the primary config ini' {
+                            $combined = $test + $optionalParamatersAll + $primaryInstallOptionalSettings
+                            Set-TargetResource @combined
+                            Get-Item -LiteralPath "$IniFilePath/$IniFileName" | Should -Not -Be $null
+                        }
+                    }
+                }
+
+                Context "When Set-TargetResource for $($test.IniFileName) has incorrect parameters" {
+                    if ($test.iniFileName -eq 'InstallCas.ini')
+                    {
+                        $wrongParameters = $test.clone()
+                        $wrongParameters.Add('ManagementPoint','PRI.contoso.com')
+                        $message = 'The parameters ManagementPoint, ManagementPointProtocol, DistributionPoint,',
+                            'DistributionPointProtocol, RoleCommunicationProtocol, ClientsUsePKICertificate,',
+                        'CCARSiteServer, CASRetryInterval, WaitForCASTimeout are used only with InstallPrimarySite.'
+                        $throwreason = 'Wrong parameter for CAS'
+
+                        It "Should throw because $throwReason" {
+                            { Set-TargetResource @wrongParameters } | Should -Throw -ExpectedMessage $message
+                        }
+                    }
+                    if ($test.iniFileName -eq 'InstallPrimary.ini')
+                    {
+                        $wrongParameters = $test.Clone()
+                        $wrongParameters.CloudConnector = $true
+                        $wrongParameters.Add('CloudConnectorServer','CAS.contoso.com')
+                        $message = 'If CloudConnector is True you must provide CloudConnectorServer and UseProxy.'
+                        $throwReason = 'Missing Cloud Connector parameter'
+
+                        It "Should throw because $throwReason" {
+                            { Set-TargetResource @wrongParameters } | Should -Throw -ExpectedMessage $message
+                        }
+                    }
+
+                    It '"Should throw because missing proxy parameters' {
```

Comment:
It "should  looks like you have a ' at the beginning and ' at the end but have a " in-between there